### PR TITLE
Feature/tiers/add select tier screen

### PR DESCRIPTION
**Pull request by juliaendre**
## [APP-XXX]

- Added new module: SelectTiers
- Added service for tiers
- Added navigation for tiers
- Added "SelectTierLandingScreen", "SelectTier" and "SelectDeductible" from [figma](https://www.figma.com/design/SNVLvLq7ztclXiDqqMNXgY/Tiers-%26-Addons-(App)?node-id=341-22125&node-type=canvas&t=sKGbRivF6XpeD89d-0)

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

### CODE REVIEW

File: Projects/SelectTier/Sources/Navigation/SelectTierNavigation.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import SwiftUI
+import hCoreUI
+
+public class SelectTierNavigationViewModel: ObservableObject {
+    @Published public var isEditTierPresented = false
+    @Published public var isEditDeductiblePresented = false
+
+    var vm = SelectTierViewModel()
+
+    init() {}
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Navigation/SelectTierNavigation.swift:11]: I think we should init SelectTierNavigation with the proper **contractId** and **source** (like its going the be expected from the 
```
enum TierSourceInputType {
    CHANGE,
    TERMINATION
}
```

After you do so, you can use it to set proper values when initing SelectTierViewModel inside this class

File: Projects/SelectTier/Sources/Navigation/SelectTierNavigation.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+public struct SelectTierNavigation: View {
+    @StateObject var router = Router()
+    @StateObject var selectTierNavigationVm = SelectTierNavigationViewModel()
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Navigation/SelectTierNavigation.swift:16]: maybe switch this to @ObservaleObject and init this view with it

File: Projects/SelectTier/Sources/Service/DemoImplementation/SelectTierClientDemo.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import hCore
+import hGraphQL
+
+public class SelectTierClientDemo: SelectTierClient {
+    @Inject var octopus: hOctopus
+
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Service/DemoImplementation/SelectTierClientDemo.swift:6]: no need for octopus here and hGraphQL at all. You probably switched it with SelectTierClientOctopus xD

File: Projects/SelectTier/Sources/Service/Protocols/SelectTierClient.swift [hunk 4]
```diff
@@ -0,0 +1,3 @@
+public protocol SelectTierClient {
+    func getTier() async throws -> TierModel
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Service/Protocols/SelectTierClient.swift:2]: Would probably need info about contractId and entry point

File: Projects/SelectTier/Sources/Views/EditDeductible.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import SwiftUI
+import hCore
+import hCoreUI
+
+struct EditDeductible: View {
+    @State var selectedDeductible: String?
+    var vm: SelectTierViewModel
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/EditDeductible.swift:7]: can change this to let

File: Projects/SelectTier/Sources/Views/EditDeductible.swift [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    @EnvironmentObject var selectTierNavigationVm: SelectTierNavigationViewModel
+
+    init(
+        vm: SelectTierViewModel
+    ) {
+        self.vm = vm
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/EditDeductible.swift:13]: Also, here you can set selectedDeductible like
self._ selectedDeductible = .init...

File: Projects/SelectTier/Sources/Views/EditDeductible.swift [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                        hText(L10n.generalCancelButton)
+                    }
+
+                }
+            }
+            .sectionContainerStyle(.transparent)
+            .padding(.top, 16)
+        }
+        .onAppear {
+            self.selectedDeductible = vm.selectedDeductible?.title ?? vm.deductibles.first?.title
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/EditDeductible.swift:67]: If we set value on init, there is not need for this.

File: Projects/SelectTier/Sources/Views/EditTier.swift [hunk 8]
```diff
@@ -0,0 +1,74 @@
+import SwiftUI
+import hCore
+import hCoreUI
+
+struct EditTier: View {
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/EditTier.swift:5]: Same parts as for the EditDeductibleView

File: Projects/SelectTier/Sources/Views/EditDeductible.swift [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import SwiftUI
+import hCore
+import hCoreUI
+
+struct EditDeductible: View {
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/EditDeductible.swift:5]: Can we rename this to _EditDeductibleView_ ?

File: Projects/SelectTier/Sources/Views/SelectTierLandingScreen.swift [hunk 10]
```diff
@@ -0,0 +1,170 @@
+import SwiftUI
+import hCore
+import hCoreUI
+import hGraphQL
+
+struct SelectTierLandingScreen: View {
+    var vm: SelectTierViewModel
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/SelectTierLandingScreen.swift:7]: Can we change this to ObservedObject var vm: SelectTierViewModel ?

File: Projects/SelectTier/Sources/Models/TierModel.swift [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+            return "449"
+        case .max:
+            return "799"
+        case .none:
+            return nil
+        }
+    }
+}
+
+public struct Deductible: Codable, Equatable, Hashable, Identifiable {
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Models/TierModel.swift:64]: Deductables will be embeded inside TierQuote and will have only values for now. Not sure would we need a description and we will see how we will add it if needed.

File: Projects/SelectTier/Sources/Views/SelectTierLandingScreen.swift [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+        .sectionContainerStyle(.transparent)
+    }
+}
+
+public class SelectTierViewModel: ObservableObject {
+    @Inject var service: SelectTierClient
+    var insuranceDisplayName: String?
+    var streetName: String?
+    var tiers: [Tier] = []
+    var deductibles: [Deductible] = []
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/SelectTierLandingScreen.swift:126]: Its probably going to be tierQuotes here

File: Projects/SelectTier/Sources/Views/SelectTierLandingScreen.swift [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+    @Published var selectedTier: Tier?
+    @Published var selectedDeductible: Deductible?
```
Comment (sladan-hedvig) [inline Projects/SelectTier/Sources/Views/SelectTierLandingScreen.swift:128]: Its probably going to be tierQuote here

**Review by sladan-hedvig — COMMENTED** [review]
Probably something I missed and needs updates somewhere else, but I think we should treat  deductibles as a thing related to the specific tier.

File: Projects/SelectTier/Sources/Models/TierModel.swift [hunk 14]
```diff
@@ -0,0 +1,69 @@
+import hGraphQL
+
+public struct TierModel: Codable, Equatable, Hashable {
```
[the remark this answers is not in the document]
Comment (juliaendre) [reply Projects/SelectTier/Sources/Models/TierModel.swift:3]: Agree!

File: Projects/App/Project.swift [hunk 15]
```diff
@@ -92,6 +92,7 @@ let appDependencies: [TargetDependency] = [
         .project(target: "Authentication", path: .relativeToRoot("Projects/Authentication")),
         .project(target: "EditCoInsured", path: .relativeToRoot("Projects/EditCoInsured")),
         .project(target: "EditCoInsuredShared", path: .relativeToRoot("Projects/EditCoInsuredShared")),
+        .project(target: "SelectTier", path: .relativeToRoot("Projects/SelectTier")),
```
[the remark this answers is not in the document]
Comment (juliaendre) [reply Projects/App/Project.swift:95]: I don't. lets change it to "ChangeTier"

File: Projects/SelectTier/Sources/Models/TierModel.swift [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+    let id: String
+    let insuranceDisplayName: String
+    let streetName: String
```
[the remark this answers is not in the document]
Comment (juliaendre) [reply Projects/SelectTier/Sources/Models/TierModel.swift:6]: Is already considered in next PR.

File: Projects/SelectTier/Sources/Models/TierModel.swift [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+    let currentPremium: MonetaryAmount
+    let newPremium: MonetaryAmount
+    let tiers: [Tier]
+    let deductibles: [Deductible]
+}
+
+public enum Tier: Codable, Equatable, Hashable, Identifiable {
```
[the remark this answers is not in the document]
Comment (juliaendre) [reply Projects/SelectTier/Sources/Models/TierModel.swift:13]: Is already considered in next PR.

File: Projects/SelectTier/Sources/Models/TierModel.swift [hunk 18]
```diff
@@ -0,0 +1,69 @@
+import hGraphQL
+
+public struct TierModel: Codable, Equatable, Hashable {
+    let id: String
+    let insuranceDisplayName: String
```
[the remark this answers is not in the document]
Comment (juliaendre) [reply Projects/SelectTier/Sources/Models/TierModel.swift:5]: Is already considered in next PR.

**Pull request closed without merging**